pydicom/charset.py

```diff
@@ -95,6 +96,167 @@
                      'iso_ir_58')
 
 
+def _encode_to_jis_x_0201(value, errors='strict'):
+    """Convert a unicode string into JIX X 0201 byte string using shift_jis
+    encodings.
+    shift_jis is a superset of jis_x_0201. So we can regard the encoded value
+    as jis_x_0201 if it is single byte character.
+
+    Parameters
+    ----------
+    value : text type
+        The unicode string as presented to the user.
+    errors : str
+        The behavior of a character which could not be encoded. If 'strict' is
+        passed, raise an UnicodeEncodeError. If others are passed, replace
+        illegal character to '?'.
+
+    Returns
+    -------
+    byte string
+        The encoded string. If some characters in value could not be encoded to
+        JIS X 0201, and `errors` is not set to 'strict', they are replaced to
+        '?'.
+
+    Raises
+    ------
+    UnicodeEncodeError
+        If errors is set to 'strict' and `value` could not be encoded with
+        JIX X 0201.
+    """
+
+    # If errors is not strict, this function is used in fallback.
+    # So keep the tail escape sequence of encoded for backward compatibility.
+    if errors != 'strict' or value == '':
+        return value.encode('shift_jis', errors=errors)
+
+    Encoder = codecs.getincrementalencoder('shift_jis')
+    encoder = Encoder()
+
+    encoded = encoder.encode(value[0])
+    if len(encoded) != 1:
+        raise UnicodeEncodeError(
+            'shift_jis', value, 0, len(value), 'illegal multibyte sequence')
+
+    msb = ord(encoded) & 0x80
+    for i, c in enumerate(value[1:], 1):
+        try:
+            b = encoder.encode(c)
+        except UnicodeEncodeError as e:
+            e.start = i
+            e.end = len(value)
+            raise e
+        if len(b) != 1 or ((ord(b) & 0x80) ^ msb) != 0:
+            character_set = 'ISO IR 14' if msb == 0 else 'ISO IR 13'
+            msg = 'Given character is out of {}'.format(character_set)
+            raise UnicodeEncodeError('shift_jis', value, i, len(value), msg)
+        encoded += b
+
+    return encoded
+
+
+def _encode_to_jis_x_0208(value, errors='strict'):
+    """Convert a unicode string into JIX X 0208 byte string using iso2022_jp
+    encodings.
+    The escape sequence which is located at the end of the encoded value have
```

Comment:
-> varies depending on value 1 ...
also: append _the_ correct escape sequence ...
(sorry for the nitpicking :) )

Reply:
Thanks reviewing thoroughly. Your comment is exactly correct. But I think that my mistake is using not 'has' but 'have'. So I think the following sentence is correct. Is it correct?

'The escape sequence which is located at the end of the encoded value has to vary depending on ... append the correct escape sequence ... '

Reply:
Yes, this is correct :)

Reply:
I fixed.
https://github.com/pydicom/pydicom/blob/cedde3962e03aa174ea1d2d5428c1eebd7063544/pydicom/charset.py#L172